src/main/java/com/company/binarysearch/PainterPartitionProblem.java

```diff
@@ -0,0 +1,88 @@
+package com.company.binarysearch;
+
+public class PainterPartitionProblem {
+    public long minTime(int[] arr, int n, int k) {
+        int max = Integer.MIN_VALUE;
+        int sum = 0;
+        for (int elem : arr) {
+            if (elem > max) {
+                max = elem;
+            }
+            sum += elem;
+        }
+        if (k == 1) {
+            return sum;
+        }
+        if (k == arr.length) {
+            return max;
+        }
+        int left = max;
+        int right = sum;
+        int answer = 0;
+        while (left <= right) {
+            int mid = left + (right - left) / 2;
+            int total = arr[0];
```

Comment:
Update the changes here also similar to first problem.

Reply:
changed